py/desispec/scripts/processingtable.py

```diff
@@ -0,0 +1,148 @@
+
+    if verbose:
+        print(f'Nights: {nights}')
+
+    ## Make the save directory exists
+    os.makedirs(exp_table_path, exist_ok=True)
+
+    ## Make the save directory if it doesn't exist
+    if not os.path.isdir(proc_table_path):
+        print(f'Creating directory: {proc_table_path}')
+        os.makedirs(proc_table_path)
+
+    ## Create an astropy table for each night. Define the columns and datatypes, but leave each with 0 rows
+    combined_table = Table()
+
+    ## Loop over nights
+    for night in nights:
+        if verbose:
+            print(f'Processing {night}')
+        exptab_name = get_exposure_table_name(night=night, extension=exp_filetype)
+        month = night_to_month(night)
+        exptable = load_table(pathjoin(exp_table_path,month,exptab_name), process_mixins=False)
+
+        if night == nights[0]:
+            combined_table = exptable.copy()
+        else:
+            # for col in exptable.colnames:
+            #     if col in combined_table.colnames:
+            #         newtype = exptable[col].dtype
+            #         curtype = combined_table[col].dtype
+            #         if newtype != curtype:
+            #             print(night, col, curtype, newtype)
+            #             if newtype == object:
+            #                 combined_as_list = np.array([[val] for val in combined_table[col]])
+            #                 print(combined_table[col])
+            #                 print(combined_as_list)
+            #                 combined_table.replace_column(col, Table.Column(name=col,data=combined_as_list,dtype=object))
+            #             elif curtype == object:
+            #                 exp_as_list = np.array([[val] for val in exptable[col]])
+            #                 print(exptable[col])
+            #                 print(exp_as_list)
+            #                 exptable.replace_column(col, Table.Column(name=col,data=exp_as_list,dtype=object))
+            #             newtype = exptable[col].dtype
+            #             curtype = combined_table[col].dtype
+            #             print(night, col, curtype, newtype)
+            #             print(exptable[col])
+            #             print(combined_table[col])
+            combined_table = vstack([combined_table, exptable])
+
+    processing_table, unprocessed_table = exptable_to_proctable(combined_table, science_types=science_types,
+                                                                joinsymb=joinsymb)
+
+    ## Save the tables
+    proc_name = get_processing_table_name(extension=prod_filetype)
+    unproc_name = proc_name.replace('processing', 'unprocessed')
+    for tab, name in zip([processing_table, unprocessed_table], [proc_name, unproc_name]):
+        if len(tab) > 0:
+            pathname = pathjoin(proc_table_path, name)
+            write_table(tab, pathname, overwrite=overwrite_files)
+            print(f'Wrote file: {name}')
+
+
+
+if __name__ == '__main__':
```

Comment:
see comments in scripts/exposuretables.py about whether this `if __name__ == '__main__':` block is deprecated testing/development code.

Reply:
Removed